cognite/client/_api/files.py

```diff
@@ -611,19 +612,50 @@ def retrieve_download_urls(
         results = tasks_summary.joined_results(unwrap_fn=lambda res: res.json()["items"])
         return {result.get("id") or result["externalId"]: result["downloadUrl"] for result in results}
 
+    @staticmethod
+    def _create_unique_file_names(file_names_in: list[str] | list[Path]) -> list[str]:
+        """
+        Create unique file names by appending a number to the base file name.
+        """
+        file_names: list[str] = [str(file_name) for file_name in file_names_in]
+        unique_original = set(file_names)
+        unique_created: list = []
+        original_count: dict = defaultdict(int)
```

Comment:
Does `mypy` complain if you don't type annotate these? Just using list and dict means items are `Any`, would be better to either remove and let it infer, or if that fails, add subtype.

Also, there's `typing.DefaultDict` for a more precise annotation.

Reply:
MyPy accepts inference on the list, but not the default-dict. Ruff changed the typing.DefaultDict to use the defaultdict-class directly...

Reply:
Nice! `ruff` ❤️